builder-base/buildkit.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+# Copyright 2020 Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Script to perform buildkit builds in daemonless mode
+
+set -e
+set -o pipefail
+set -x
+
+tmp=$(mktemp -d /tmp/buildctl-daemonless.XXXXXX)
+trap "kill \$(cat $tmp/pid); rm -rf $tmp" EXIT
+
+start::buildkitd() {
+    buildkitd --addr=unix:///run/buildkit/buildkitd.sock --oci-worker-platform=linux/amd64 --oci-worker-platform=linux/arm64 >$tmp/log 2>&1 &
```

Comment:
Are you missing the flag defined here? https://github.com/aws/eks-distro-prow-jobs/blob/main/scripts/buildkitd-entrypoint.sh#L21

Reply:
We can use that flag only if the command is invoked with Rootlesskit. Within the security context of a single container, it is complicated to allow only the buildkit daemon to run as a rootless user with the rest of the build operations as root. Had a discussion with Vignesh and Micah about this, and we reached a consensus that we will not be using rootless here. Also these builds are ephemeral and isolated, and the script has logic to kill the daemon even if the build fails, so it should be okay to run as root.